src/probnum/_config.py

```diff
@@ -0,0 +1,83 @@
+import contextlib
+from typing import Any
+
+
+class Configuration:
+    """
+    Configuration over which some mechanics of probnum can be controlled dynamically.
+
+    ``probnum`` provides some configurations together with default values. These
+    are listed in the tables below.
+    Additionally, the user can register own configuration entries via the method
+    :meth:`register`. Configuration entries can only be registered once and can only
+    be used (accessed or overwritten) once they have been registered.
+
+    - ``probnum.randvars``
+
+    +----------------------------------+-----------------------------------------------+
+    | Config entry                     | Description                                   |
+    +==================================+===============================================+
+    | ``covariance_inversion_damping`` | A (typically small) value that is per         |
+    |                                  | default added to the diagonal of covariance   |
+    |                                  | matrices in order to make inversion           |
+    |                                  | numerically stable.                           |
+    +----------------------------------+-----------------------------------------------+
+    | ``...``                          | ...                                           |
+    +----------------------------------+-----------------------------------------------+
```

Comment:
I dont understand how/why a subpackage (randvars) is explicitly mentioned here. Is it because this parameter is only used there? Is it because it needs to be accessed with `probnum.config.randvars` or something like this? Would be great if you could explain this to me :)

Reply:
That's not strictly necessary. I think we agreed on module-wide configs when discussing this PR, so I mirrored that in the documentation, as well. We can readily change this decision in favor of `ProbNum`-wide configs/settings.

Reply:
I think this relates to the other discussion above, so I wont repeat my questions/confusions here :)